lib/tty/prompt/answers_collector.rb

```diff
@@ -28,12 +29,15 @@ def call(&block)
       #   key(:name).ask('Name?')
       #
       # @api public
-      def key(name, &block)
+      def key(name)
         @name = name
-        if block
-          answer = create_collector.(&block)
-          add_answer(answer)
-        end
+        collect_block(&Proc.new) if block_given?
```

Comment:
That talk is referring to really old Ruby versions and I would be surprised if the difference persist. Will need to ask Aaron Paterson 😄  My preference would be to use explicit `&block` parameter and here is why, this is a CLI tool, unless we're talking about software system calling this inordinate amount of times I sway towards aesthetics rather than optimisation.